export_job/export_zip_from_manifest_job.py

```diff
@@ -0,0 +1,184 @@
+import aiohttp
+import asyncio
+import json
+import os
+import shutil
+from gen3.tools.download.manifest import (
+    describe_access_to_files_in_workspace_manifest,
+    download_files_in_workspace_manifest,
+    list_files_in_workspace_manifest,
+)
+from gen3.auth import Gen3Auth
+import requests
+import boto3
+
+from temporary_api_key import TemporaryAPIKey
+
+
+CHUNK_SIZE = 10
+MANIFEST_FILENAME = "manifest.json"
+EXPORT_DIR = "export"
+OUTPUT_PREFIX = "[out] "
+MAX_DOWNLOAD_SIZE = 250000000
+DEFAULT_ERROR_MESSAGE = (
+    "Unable to complete this download. Please try again later, or use the Gen3 client."
+)
+
+
+async def build_manifest_from_study_ids(hostname, token, study_ids):
+    """
+    build a manifest from a list of metadata guids representing study ids
+    dumps result to gen3-sdk compatible manifest file
+    """
+    ongoing_requests = []
+    study_metadata = []
+
+    print(f"Assembling manifest for study ids: {study_ids}")
+
+    async with aiohttp.ClientSession() as client:
+        for study_id in study_ids:
+            ongoing_requests += [
+                client.get(
+                    f"https://{hostname}/mds/aggregate/metadata/guid/{study_id}",
+                    headers={"Authorization": f"bearer {token}"},
+                )
+            ]
+            if len(ongoing_requests) == CHUNK_SIZE:
+                study_metadata += await asyncio.gather(
+                    *[req.json() for req in await asyncio.gather(*ongoing_requests)]
+                )
+
+        study_metadata += await asyncio.gather(
+            *[req.json() for req in await asyncio.gather(*ongoing_requests)]
+        )
+
+    # print(f"Retrieved metadata: {study_metadata}")
+    manifest = []
+    for study in study_metadata:
+        if not study["__manifest"]:
```

Comment:
Do we make it configurable which field name we use? Is it `__manifest` for some and `__files` for others? Not saying it's good if it's configurable, but just checking. I wonder if we need to query the `gitops` info to see what the field name is

Reply:
For HEAL, I think it's always been `__manifest` : https://github.com/uc-cdis/cdis-manifest/blob/369b80452adb4a91fdef7fd1b6f2e283c12168e6/healdata.org/portal/gitops.json#L126
At least that's how we've been building manifests from the UI

But I can change this to be passed as an env var from sower, not sure if there's a way to avoid that being a duplicate configuration

Reply:
You could use put some bash script in your cloud-auto PR to extract that from portal config and put it into a k8s configmap or secrets

Reply:
If I search across our repos I only see us ever setting `manifestFieldName` to `__manifest`. I think the best thing to do would be to _not_ have that field configurable, keep it hard-coded here, and take it out of configurations. I think making that field configurable just adds more headache especially if it's consistent everywhere

Reply:
Yeah I remoember I have mention to have a set of `pre-defined` field name rules for MDS data that serves as a convention enforced over all MDS

This way we can reduce this kind of configuration complexity by a lot